packages/dev/serializers/src/glTF/2.0/glTFAnimation.ts

```diff
@@ -578,44 +596,110 @@ export class _GLTFAnimation {
 
             // create bufferview and accessor for keyed values.
             outputLength = animationData.outputs.length;
-            byteLength = _GLTFUtilities._GetDataAccessorElementCount(dataAccessorType) * 4 * animationData.outputs.length;
+            byteLength = getAccessorElementCount(dataAccessorType) * 4 * animationData.outputs.length;
 
             // check for in and out tangents
-            bufferView = _GLTFUtilities._CreateBufferView(0, binaryWriter.getByteOffset(), byteLength, undefined, `${name}  data view`);
+            bufferView = createBufferView(0, binaryWriter.byteOffset, byteLength);
             bufferViews.push(bufferView);
 
+            const rotationQuaternion = new Quaternion();
+            const eulerVec3 = new Vector3();
+            const position = new Vector3();
+            const tempQuaterionArray = [0, 0, 0, 0];
+            const isCamera = babylonTransformNode instanceof Camera;
+
             animationData.outputs.forEach(function (output) {
-                output.forEach(function (entry) {
-                    binaryWriter.setFloat32(entry);
-                });
+                if (convertToRightHanded) {
+                    switch (animationChannelTargetPath) {
+                        case AnimationChannelTargetPath.TRANSLATION:
+                            Vector3.FromArrayToRef(output, 0, position);
+                            convertToRightHandedPosition(position);
+                            binaryWriter.writeFloat32(position.x);
+                            binaryWriter.writeFloat32(position.y);
+                            binaryWriter.writeFloat32(position.z);
+                            break;
+
+                        case AnimationChannelTargetPath.ROTATION:
+                            if (output.length === 4) {
+                                Quaternion.FromArrayToRef(output, 0, rotationQuaternion);
+                            } else {
+                                Vector3.FromArrayToRef(output, 0, eulerVec3);
+                                Quaternion.FromEulerVectorToRef(eulerVec3, rotationQuaternion);
+                            }
+
+                            if (isCamera) {
+                                convertCameraRotationToGLTF(rotationQuaternion);
+                            } else {
+                                if (!Quaternion.IsIdentity(rotationQuaternion)) {
+                                    convertToRightHandedRotation(rotationQuaternion);
+                                }
+                            }
+
+                            rotationQuaternion.toArray(tempQuaterionArray);
+                            binaryWriter.writeFloat32(tempQuaterionArray[0]);
+                            binaryWriter.writeFloat32(tempQuaterionArray[1]);
+                            binaryWriter.writeFloat32(tempQuaterionArray[2]);
+                            binaryWriter.writeFloat32(tempQuaterionArray[3]);

```

Comment:
Is this necessary? Does writing the components of the quaternion directly not work for some reason?

```suggestion
                            binaryWriter.writeFloat32(rotationQuaternion.x);
                            binaryWriter.writeFloat32(rotationQuaternion.y);
                            binaryWriter.writeFloat32(rotationQuaternion.z);
                            binaryWriter.writeFloat32(rotationQuaternion.w);
```